### ISSUE TITLE
Does the package has types?

### ISSUE THREAD

**Issue by JooZef315** [opendatacam/node-moving-things-tracker#47]
I am trying to use it but it seems that it doesn't has types, or I am missing something?

ERROR: Could not find a declaration file for module 'node-moving-things-tracker'. 'node_modules/node-moving-things-tracker/main.js' implicitly has an 'any' type.

**Comment by vsaw** [comment]
That is correct. It is written in plain JS without type declarations. If you happen to add type declarations I'm happy to merge them upstream :)

**Comment by JooZef315** [comment]
Hi, I did added types for example methods for now, but before continue working on it i want to ask you about if I could get many instances of the tracker instead of being a singleton? if you could guide me with main steps to do so I could add this as well, or is it possible already with current structure?

**Comment by vsaw** [comment]
It is not possible with the current code base. 

Implementing it should be fairly straightforward though. 

At the moment all methods are exported directly and all variables to track state are global. 

To allow for multiple instances of the tracker all you need to do is wrap everything into an object and then export the entire object. That should allow you to create multiple instances.
